design/CONSUMER.md

```diff
@@ -8,4 +8,28 @@ Things to consider:
 - persisting the last seen page
 - mapping events to new data/actions
 
-This library should not try to do too much but focus on tools to make it easier to fetch events and action on them.
\ No newline at end of file
+This library should not try to do too much but focus on tools to make it easier to fetch events and action on them.
+
+## How
+
+```mermaid
+sequenceDiagram
+    participant C as Consumer
+    participant S as Consumer State
+    participant P as Producer
+    C->>+P: Fetch meta data
+    P-->>-C: Return meta data
+    
+    loop Until consumer is shut down
+        C->>+S: Fetch last stored page
+        S-->>-C: (Last seen page) OR nothing
+        C->>+P: Request page of events
+        P-->-C: Page of events
+        C-->>+S: Save intents of events
```

Comment:
Would it make sense to use the 'command' naming instead of intent? It feels more familiar to me and I think would prevent having to explain what an intent is. Are there cases where the intent is not a command? Is it useful/different enough to warrant it's own concept?

Reply:
There are different from my point of view. Intent could wrap up multiple commands in multiple systems. I mean you could say that is a command on the client-side. I see command as a pattern, intent as an idea.